src/main/java/com/tabnine/binary/BinaryProcessGateway.kt

```diff
@@ -12,12 +18,40 @@ open class BinaryProcessGateway {
 
     @Throws(IOException::class)
     open fun init(command: List<String?>?) {
-        val createdProcess = ProcessBuilder(command).start()
+        val pb = ProcessBuilder(command)
+        if (instance.useIJProxySettings) {
+            setProxyEnvironmentVariables(pb)
+        }
+        val createdProcess = pb.start()
 
         process = createdProcess
         reader = BufferedReader(InputStreamReader(createdProcess.inputStream, StandardCharsets.UTF_8))
     }
 
+    private fun setProxyEnvironmentVariables(pb: ProcessBuilder) {
+        val env = pb.environment()
+        val serverUrl = StaticConfig.getBundleServerUrl()
+        if (serverUrl.isPresent) {
+            val serverUrlString = serverUrl.get()
+            val tabnineServerProxy = CommonProxy.getInstance().select(URL(serverUrlString)).firstOrNull()
+            if (tabnineServerProxy?.type() == Proxy.Type.DIRECT) {
+                env["NO_PROXY"] = serverUrlString
+                env["no_proxy"] = serverUrlString
+            }
```

Comment:
extract to function `setOurServerUrlProxy`

Reply:
this should be the host, right? not including protocol